src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookCause.java

```diff
@@ -0,0 +1,71 @@
+package io.jenkins.plugins.gitlabbranchsource;
+
+import hudson.markup.EscapedMarkupFormatter;
+import hudson.triggers.SCMTrigger.SCMTriggerCause;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.gitlab4j.api.webhook.MergeRequestEvent;
+import org.gitlab4j.api.webhook.MergeRequestEvent.ObjectAttributes;
+import org.gitlab4j.api.webhook.PushEvent;
+import org.gitlab4j.api.webhook.TagPushEvent;
+
+public class GitLabWebHookCause extends SCMTriggerCause {
+
+    private String description;
+
+    public GitLabWebHookCause() {
+        super("");
+    }
+
+    public GitLabWebHookCause fromPush(PushEvent pushEvent) {
+        String userName = pushEvent.getUserName();
+        if (StringUtils.isBlank(userName)) {
+            description = Messages.GitLabWebHookCause_ShortDescription_Push_noUser();
+        } else {
+            description = Messages.GitLabWebHookCause_ShortDescription_Push(userName);
+        }
+        return this;
+    }
+
+    public GitLabWebHookCause fromMergeRequest(MergeRequestEvent mergeRequestEvent) {
+        ObjectAttributes objectAttributes = mergeRequestEvent.getObjectAttributes();
+        String id = String.valueOf(objectAttributes.getIid());
+        String sourceNameSpace = objectAttributes.getSource().getNamespace();
+        String targetNameSpace = objectAttributes.getTarget().getNamespace();
+        String namespace =
```

Comment:
I suggest either you keep `namespace` camel cased or don't.

Reply:
I am not modifying the user typed name space.
and I am only using it to compare if source and target is the same